src/server/hset_family.cc

```diff
@@ -720,19 +721,35 @@ OpResult<vector<long>> OpHExpire(const OpArgs& op_args, string_view key, uint32_
   return HSetFamily::SetFieldsExpireTime(op_args, ttl_sec, key, values, pv);
 }
 
-// HSETEX key [NX] tll_sec field value field value ...
+OpSetParams LoadHSetExParams(CmdArgParser& parser) {
+  OpSetParams op_sp{};
+  while (true) {
+    if (parser.Check("NX")) {
+      op_sp.skip_if_exists = true;
+    } else if (parser.Check("KEEPTTL")) {
+      op_sp.keepttl = true;
+    } else {
+      break;
+    }
+  }
+
```

Comment:
Thinking about this again why don't we just enforce an order ? `NX` should go first, followed by `KEEPTTL` if it exists.

So we rewrite this as:

```
if (parser.Check("NX")) {
  op_sp.skip_i_exists = true;
}
if (parser.Check("KEEPTTL")) {
      op_sp.keepttl = true;
}
```

And we are done.

Reply:
change to something similar

Reply:
I dont think this is a good decision to enforce the order. We dont do this in other commands afaik. This is actually more of a product decision of to enforce order of commands.
Please create a fix PR to not enforce the order @abhijat